### DISCUSSION TITLE
Keep deleted issue author as ghost

### DISCUSSION THREAD

**Discussion by chinesedfan** [community/community#24888] [Apps, API and Webhooks]
<p>Now if the user account is deleted, GraphQL returns <code>null</code>, while REST v3 returns <code>ghost</code>. For example,</p>
<pre><code class="lang-auto">
query {  
&amp;nbsp; &amp;nbsp;repository(owner: "andrew-d", name: "static-binaries") {  
&amp;nbsp; &amp;nbsp; &amp;nbsp;issue(number: 22) {  
&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;author {  
&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;login  
&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;}  
&amp;nbsp; &amp;nbsp; }  
&amp;nbsp; }  
}

</code></pre>
<p>Compared with, <code>[https://api.github.com/repos/andrew-d/static-binaries/issues/22](https://api.github.com/repos/andrew-d/static-binaries/issues/22)</code>.</p>

**Comment by lee-dohm** [comment]
<p>Thanks for the feedback. I’m investigating this and will get back to you as soon as I have more information.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>In talking with the developers, this was a deliberate design decision to make this change for the GraphQL API. For example, imagine this query:</p>
<pre><code>query {
   repository(owner: "foo", name: "bar") {
     issue(number: 22) {
       author {
         ...on User {
          repositories(first: 10) {
            nodes {
              nameWithOwner
            }
          }
        }
      }
    }
  }
}
</code></pre>
<p>By returning a <code>User.ghost</code> for the issue author in this case, we then present a bunch of data about the ghost user (or no data), which is (at best) useless to the user, and at worst will lead them to mis-associate information.</p>
<p>Instead, by returning <code>null</code>, we are able to indicate exactly where in the graph there is no longer valid information.</p>
<p>I hope that helps!</p>
